code/game/gamemodes/miniantags/pulsedemon/pulsedemon.dm

```diff
@@ -0,0 +1,821 @@
+		// returns used energy, not damage dealt, but ez conversion with /20
+		dealt = electrocute_mob(L, current_cable.powernet, src, siemens_coeff) / 20
+	else if(charge >= 1000)
+		dealt = L.electrocute_act(30, src, siemens_coeff)
+		adjust_charge(-1000)
+	if(dealt > 0)
+		do_sparks(rand(2, 4), FALSE, src)
+	add_attack_logs(src, L, "shocked ([dealt] damage)")
+
+/mob/living/simple_animal/demon/pulse_demon/proc/is_under_tile()
+	var/turf/T = get_turf(src)
+	return T.transparent_floor || T.intact || HAS_TRAIT(T, TRAIT_TURF_COVERED)
+
+// cable (and hijacked APC) view helper
+/mob/living/simple_animal/demon/pulse_demon/proc/update_cableview()
+	if(!client)
+		return
+
+	// clear out old images
+	for(var/image/current_image in cable_images + apc_images)
+		client.images -= current_image
+
+	var/turf/T = get_turf(src)
+
+	// regenerate for all cables on our (or our holder's) z-level
+	cable_images.Cut()
+	for(var/datum/regional_powernet/P in SSmachines.powernets)
+		for(var/obj/structure/cable/C in P.cables)
+			var/turf/cable_turf = get_turf(C)
+			if(T.z != cable_turf.z)
+				break // skip entire powernet if it's off z-level
+
+			var/image/cable_image = image(C, C, layer = ABOVE_LIGHTING_LAYER, dir = C.dir)
+			// good visibility here
+			cable_image.plane = ABOVE_LIGHTING_PLANE
+			LAZYADD(cable_images[cable_turf], cable_image)
+			client.images += cable_image
+
+	// same for hijacked APCs
+	apc_images.Cut()
+	for(var/obj/machinery/power/apc/A in hijacked_apcs)
+		var/turf/apc_turf = get_turf(A)
+		if(T.z != apc_turf.z)
+			continue
+		// parent of image is the APC, not the turf because of how clicking on images works
+		var/image/apc_image = image('icons/obj/power.dmi', A, "apcemag", ABOVE_LIGHTING_LAYER, A.dir)
+		apc_image.plane = ABOVE_LIGHTING_PLANE
+		LAZYADD(apc_images[apc_turf], apc_image)
+		client.images += apc_image
+
+/mob/living/simple_animal/demon/pulse_demon/emp_act(severity)
+	if(emp_debounce)
+		return
+
+	. = ..()
+	visible_message("<span class ='danger'>[src] [pick("fizzles", "wails", "flails")] in anguish!</span>")
+	playsound(get_turf(src), pick(hurt_sounds), 30, TRUE)
+	throw_alert(ALERT_CATEGORY_NOREGEN, /obj/screen/alert/pulse_noregen)
+	switch(severity)
+		if(EMP_LIGHT)
+			adjustHealth(round(max(initial(health) / 4, round(maxHealth / 8))))
+			regen_lock = 3
+		if(EMP_HEAVY)
+			adjustHealth(round(max(initial(health) / 3, round(maxHealth / 6))))
+			regen_lock = 5
+	emp_debounce = TRUE
+	addtimer(VARSET_CALLBACK(src, emp_debounce, FALSE), 0.1 SECONDS, TIMER_UNIQUE | TIMER_OVERRIDE)
+
+/mob/living/simple_animal/demon/pulse_demon/proc/try_attack_mob(mob/living/L)
+	if(!is_under_tile() && L != src)
+		do_attack_animation(L)
+		try_shock_mob(L)
+
+/mob/living/simple_animal/demon/pulse_demon/UnarmedAttack(atom/A)
+	if(isliving(A))
+		try_attack_mob(A)
+	else if(isitem(A) && !is_under_tile())
+		var/obj/item/O = A
+		var/obj/item/stock_parts/cell/C = O.get_cell()
+		if(C && C.charge)
```

Comment:
```suggestion
		if(C?.charge)
```

Reply:
i refuse to use ?. also it passes 0

Reply:
maybe make the first part istype()? that might help too

Reply:
actually